### PR TITLE
Sv 107/crm routes

### PR DESCRIPTION
**Pull request by ccp92**
# What?

Adds routes to deal with user mapping between CRM and Google Auth

# Why?

Creates a link between google authentication and users stored in CRM which should allow us to expand the google auth to the legacy process apps to ensure a seamless login experience

# Checklist

- [x] Unit tests
- [ ] Integration tests
- [ ] Cypress tests

# Next Steps
- Look to link this new table into the existing ITV/THC/HC apps

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -45,3 +51,43 @@ MaTApp.getInitialProps = async (context: AppContext) => {
...
+    if (existingUserMapping) {
+      return;
+    } else {
+      const crmUserGuid = await getCrmUserGuid(hackneyToken.email);
+      let crmId;
+      if (crmUserGuid === undefined) {
+        const splitName = hackneyToken.name.split(' ')
+        const user = {
+          fullName: hackneyToken.name,
+          firstName: splitName[0],
```
Comment (LBHTKarki) [inline src/pages/_app.tsx:77]: this is probably best we can do with the first and last name for now

File: src/gateways/crmGateway.ts [hunk 2]
```diff
@@ -143,21 +152,27 @@ class CrmGateway implements CrmGatewayInterface {
...
-      put(``, {
+      post(`${process.env.CRM_API_URL}/api/data/v8.2/hackney_estateofficers`, crmUser, {
         headers: {
           "Authorization": `Bearer ${this.crmApiToken.token}`,
           "Prefer": "odata.include-annotations=\"OData.Community.Display.V1.FormattedValue\""
         }
       })
       .then((response) => {
-        const data = response.data;
+        const data = response.headers.location.match(/\((.*?)\)/)[1];
```
Comment (LBHTKarki) [inline src/gateways/crmGateway.ts:175]: I presume the response headers are static for this particular call, so this will always find the correct value?

**Review by LBHTKarki — APPROVED** [review]
Looks good. Not much we can do about making the first name/last name handling more robust unless the cookie gets updated with those values.

File: src/gateways/crmGateway.ts (hunk 2 above)
Comment (ccp92) [reply src/gateways/crmGateway.ts:175]: Yeah, it's a bit frustrating that it's not part of a proper response body, but the created user's guid is stored inside a URL in the header each time.

File: src/pages/_app.tsx (hunk 1 above)
Comment (ccp92) [reply src/pages/_app.tsx:77]: Yeah. I've done the split and call in a way that it should still pick up the correct last name, even if they have their middle name listed in the cookie.

**Pull request merged**